cloudformation/jolly-roger.yaml

```diff
@@ -136,11 +136,23 @@ Parameters:
     Type: String
     Default: "true"
     AllowedValues: ["true", "false"]
+  EnableSingleInstanceMode:
+    Description: If true, disable the NLB and point the domain directly at a single instance. The TLS certificate will be automatically managed via Let's Encrypt - be sure to accept their terms of service.
+    Type: String
+    Default: "false"
+    AllowedValues: ["true", "false"]
```

Comment:
Should this just be a tristate? `ServingMode` with `AllowedValues: ["NLB", "Single", "None"]` or something like that? It feels slightly weird to be able to toggle the two independently.

Reply:
Yeah, that makes perfect sense. Done.

I was also thinking about whether to do anything different with AppMinSize/AppMaxSize/AppDesiredCapacity, since some values there don't make sense in single instance mode. But short of some sort of validation rule (is that even possible? Maybe with a lambda?) I didn't think of anything that seemed worthwhile.

Reply:
Yeah I think that's true, but I don't think there's any reasonable way to do that validation. In theory, you could do it at application time with a Lambda, but then you'd get into rolling the stack forwards and backwards and that's always been a little messy with CFN. I don't think there's any way to do it as a pre-application check. I think it's fine to say that that's pushing beyond the limits of what CFN can do.